charmdir.go

```diff
@@ -120,11 +120,9 @@ func ReadCharmDir(path string) (dir *CharmDir, err error) {
 			return nil, errors.New("invalid revision file")
 		}
 	}
-	version, _, err := dir.MaybeGenerateVersionString(logger)
-	if err != nil {
-		// We don't want to stop, even if the version cannot be generated
-		logger.Errorf("unexpected problem trying to generate version string: %q", err)
-	}
+
+	var dummyLogger = NopLogger{}
+	version, _, err := dir.MaybeGenerateVersionString(dummyLogger)
```

Comment:
I suspected that the past change here can lead to a high volume of logging, as a charm can be read more than it will be "archived"